tokstr.h

```diff
@@ -0,0 +1,25 @@
+#ifndef __TOKSTR_H
+#define __TOKSTR_H
+
+// tokstr -- textual token iterator with some input independence
+// 2022-01-25 [initially released inside dnsdbq]
+
+/* example:
+
+	tokstr_h ts = tokstr_string("this:is+-test");
+	for (char *t; (t = tokstr_next(ts, "-:+")) != NULL; free(t))
+		printf("\t\"%s\"\n", t);
+	tokstr_last(&ts);
+
+ * will output "this", "is", and "test".
+ */
+
+struct tokstr;
+typedef struct tokstr *tokstr_t;
+
+tokstr_t tokstr_buffer(const char *source, size_t size);
```

Comment:
Needs some function comments on the variations between _buffer and _string

Reply:
see 7768f14.